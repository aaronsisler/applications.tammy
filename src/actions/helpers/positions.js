export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
