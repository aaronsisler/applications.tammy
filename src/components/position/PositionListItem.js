import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const PositionListItem = ({ id, title, jobId }) => (
    <Link className="position_list_item" to={`/position/${id}`}>
        <div>
            <h3 className="position_list_item_title">{title}</h3>
        </div>
        <h3 className="position_list_item_job_id">
            {jobId}
        </h3>
    </Link>
);

export default PositionListItem;

PositionListItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    jobId: PropTypes.string.isRequired,
};
