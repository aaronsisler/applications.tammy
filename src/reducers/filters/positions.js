const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
