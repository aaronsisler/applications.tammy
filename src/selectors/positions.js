// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
