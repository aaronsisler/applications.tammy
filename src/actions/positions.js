import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
