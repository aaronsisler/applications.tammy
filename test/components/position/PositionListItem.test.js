import React from 'react';
import { shallow } from 'enzyme';
import { PositionListItem } from 'Position/PositionListItem';
import positions from '../../fixtures/positions';

describe('PositionListItem', () => {
    const startSetPosition = jest.fn();
    const [position] = positions;
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <PositionListItem
                {...position}
                startSetPosition={startSetPosition}
            />);
    });

    it('should render PositionListItem correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call startSetPosition prop when clicked', () => {
        wrapper.find('.position_list_item').simulate('click');

        expect(startSetPosition).toHaveBeenLastCalledWith(position.positionId);
    })
});
