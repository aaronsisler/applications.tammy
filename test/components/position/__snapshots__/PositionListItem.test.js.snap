// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionListItem should render PositionListItem correctly 1`] = `
<div
  className="position_list_item"
  onClick={[Function]}
>
  <div
    className="position_list_item__content"
  >
    <div
      className="position_list_item__title"
    >
      first title
    </div>
    <div
      className="position_list_item__job_id"
    >
      Job Id: 
      first job id
    </div>
  </div>
  <div
    className="position_list_item__location"
  >
    first location
  </div>
</div>
`;
